editor/src/core/workers/utils.ts

```diff
@@ -1,10 +1,45 @@
-import { BASE_URL, URL_HASH } from '../../common/env-vars'
-import { setBranchNameFromURL } from '../../utils/branches'
-
-export function workerForFile(fileName: string): Worker {
-  let workerURL = new URL(`${BASE_URL}${fileName}`)
-  let workerQueryParams = workerURL.searchParams
-  workerQueryParams.set('hash', URL_HASH)
-  setBranchNameFromURL(workerQueryParams)
-  return new Worker(workerURL.href)
+import { BASE_URL } from '../../common/env-vars'
+
+/**
+ * Webpack 5 only supports a single public path for assets and feeding the Worker url via `import.meta.url`
+ * This causes problems when using a CDN, because of same origin restrictions, you are not allowed to load workers from cdns.
+ *
+ * The solution is to use the webpack-supported global __webpack_public_path__ variable, which updates the value returned by import.meta.url
+ *
+ * in all of these cases, oldPublicPath is the asset url (the CDN url in deployed editors). we temporarily change it to BASE_URL (which is the url the editor is loaded from)
+ * and then after loading the worker, restore it to the CDN url.
```

Comment:
> and then after loading the worker

So does this mean that all of the workers will be downloading their js bundles directly from the server rather than the CDN?

Reply:
yes, that's happening now too, because you can't eval js coming from the cdn

Reply:
Well that's thoroughly disappointing :disappointed:

Reply:
Oh that's bad times, I thought it was fine because it's a subdomain.

Reply:
It's still a vast improvement because it means the workers themselves will now be sharing chunks. It does however mean that chunks shared between the editor and the workers will still be downloaded twice, with the worker downloads being slower

Reply:
are we serving the editor js from the cdn?! I'm a bit unclear on these things. what actually triggered me doing this __webpack_public_path__ business was seeing the same origin error message when trying to load the workers on staging

Reply:
I checked just now on and we are loading it from the CDN.

Reply:
The editor loads all of its js bundles from CDN

Reply:
<img width="998" alt="image" src="https://user-images.githubusercontent.com/2226774/132699016-4849de0e-171d-4a9a-8423-d6f200a07346.png">

@Rheeseyb I just opened utopia.pizza to double check and it seems like we are loading the workers from not-the-cdn.